File: Back-End/src/main/java/com/boyslab/ads/dtos/response/request/RequestResponseDto.java
package com.boyslab.ads.dtos.response.request;
import com.boyslab.ads.entities.Request;
import com.boyslab.ads.entities.enums.Status;

public record RequestResponseDto(
        String tc,
        String name,
        String surname,
        int birthDay,
        String description,
        String phone,
        String city,
        String district,
        String neighbourhood,
        String street,
        String locationDescription,
        Status status,
        String categoryName
) {
    public static RequestResponseDto convertToDto(Request request){
        return new RequestResponseDto(
                request.getTc(),
                request.getName(),
                request.getSurname(),
                request.getBirthDay(),
                request.getDescription(),
                request.getPhone(),
                request.getCity(),
                request.getDistrict(),
                request.getNeighbourhood(),
                request.getStreet(),
                request.getLocationDescription(),
                request.getStatus(),
                request.getCategory().getName()
        );
    }

}

File: AfetProjesi/app/src/main/java/com/example/afetprojesi/dtos/responses/request/RequestResponseDto.kt
package com.example.afetprojesi.dtos.responses.request

import com.example.afetprojesi.dtos.requests.request.Status
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class RequestResponseDto(
    @Expose
    @SerializedName("tc")
    var tc : String,

    @Expose
    @SerializedName("name")
    var name : String,

    @Expose
    @SerializedName("surname")
    var surname : String,

    @Expose
    @SerializedName("birthDay")
    var birthDay : Int,

    @Expose
    @SerializedName("description")
    var description : String,

    @Expose
    @SerializedName("phone")
    var phone : String,

    @Expose
    @SerializedName("city")
    var city : String,

    @Expose
    @SerializedName("district")
    var district : String,

    @Expose
    @SerializedName("neighbourhood")
    var neighbourhood : String,

    @Expose
    @SerializedName("street")
    var street : String,

    @Expose
    @SerializedName("locationDescription")
    var locationDescription : String,
    @Expose
    @SerializedName("status")
    var status: Status,

    @Expose
    @SerializedName("categoryName")
    var categoryName : String

)
